### PR TITLE
added dynamic docker swarm nodes inventory

### PR DESCRIPTION
**Pull request by morph027**
##### SUMMARY

This adds a dynamic inventory for docker swarm nodes.

##### ISSUE TYPE

 - Feature Pull Request

##### ANSIBLE VERSION

```
2.5.0
```

### CODE REVIEW

File: contrib/inventory/docker_swarm_nodes.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# Installation
```
Comment (DBendit) [inline contrib/inventory/docker_swarm_nodes.py:20]: Could you please wrap the documentation in a `DOCUMENTATION` block, similar to https://github.com/ansible/ansible/blob/devel/contrib/inventory/docker.py?

Additionally, please split into similar sections: general info, requirements, configuration, and examples. I see that you have most of these, but making it more explicit would be helpful for future users.

**Review by DBendit — COMMENTED** [review]
Although new modules and plugins don't require changelog fragments, I don't believe dynamic inventory scripts fall into those categories - I'm double checking that in IRC now. If I'm right, then you'll need to include a changelog fragment along with this. See https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#creating-new-fragments for details.

Additionally, Ansible now supports inventory plugins as an alternative to dynamic inventory scripts. See https://docs.ansible.com/ansible/latest/plugins/inventory.html for more details there. It might be a better idea to implement this as an inventory plugin instead, as it has better integration with the rest of Ansible.

**Review by DBendit — CHANGES_REQUESTED** [review]
After checking with the folks in #ansible-devel, it seems that we're no longer accepting new inventory scripts. Please re-write this as an inventory plugin instead. See the link above for more information on that.

needs_revision

**Comment by morph027** [comment]
Okay, thanks for your feedback ;)

**Comment by gundalow** [comment]
@morph027 Hi, Are you able to rewrite this as an inventory plugin

https://docs.ansible.com/ansible/devel/dev_guide/developing_inventory.html#inventory-plugins https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/inventory/

**Comment by morph027** [comment]
Will try, if i can make up some spare time ;)

**Comment by gundalow** [comment]
@morph027 Please do shout out if you need any help with this. Their is an active Docker community at Ansible

**Comment by morph027** [comment]
I've got 2 weeks vacation next week, with sun and fun. Guess, there will be some time. Docker is no problem, just need to adjust my poor coding skills to the guidelines ;)

**Comment by morph027** [comment]
moved to https://github.com/ansible/ansible/pull/53058

**Pull request closed without merging**